devicetypes/zooz/zooz-zen52-double-relay.src/zooz-zen52-double-relay.groovy

```diff
@@ -0,0 +1,437 @@
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+*/
+
+import groovy.transform.Field
+
+@Field static Map commandClassVersions = [
+	0x20: 1,	// Basic
+	0x22: 1,	// ApplicationStatus
+	0x25: 1,	// SwitchBinary
+	0x55: 1,	// TransportService
+	0x59: 1,	// AssociationGrpInfo
+	0x5A: 1,	// DeviceResetLocally
+	0x5B: 1,	// CentralScene
+	0x5E: 2,	// ZwaveplusInfo
+	0x60: 3,	// MultiChannel
+	0x6C: 1,	// Supervision
+	0x70: 1,	// Configuration
+	0x72: 2,	// ManufacturerSpecific
+	0x73: 1,	// Powerlevel
+	0x7A: 2,	// FirmwareUpdateMd
+	0x85: 2,	// Association
+	0x86: 1,	// Version
+	0x87: 2,	// Indicator (3)
+	0x8E: 2,	// MultiChannelAssociation
+	0x98: 1,	// Security S0
+	0x9F: 1		// Security S2
+]
+
+@Field static int supervisionCC = 108
+@Field static int btnPushed = 0
+@Field static int btnReleased = 1
+@Field static int btnHeld = 2
+@Field static int mainEndpoint = 0
+@Field static List<Integer> relayEndpoints = [1, 2]
+@Field static List<String> supportedButtonValues = ["pushed","held","pushed_2x","pushed_3x","pushed_4x","pushed_5x"]
+
+@Field static Map configParams = [
+	ledIndicator: [num:2, title:"Led indicator", size:1, defaultVal:1, options:[0:"Disabled", 1:"Enabled"]],
+	sceneControl: [num:16, title:"Scene Control", size:1, defaultVal:1, options:[0:"Disabled", 1:"Enabled"]],
+	relay1AutoOff: [num:3, title:"Relay 1 Auto Off Timer", size:2, defaultVal:0, range:"0..65535", desc:"0(disabled), 1..65535(timer unit)"],
+	relay1AutoOn: [num:4, title:"Relay 1 Auto On Timer", size:2, defaultVal:0, range:"0..65535", desc:"0(disabled), 1..65535(timer unit)"],
+	relay1TimerUnit: [num:7, title:"Relay 1 Timer Unit", size:1, defaultVal:1, options:[1:"Minutes", 2:"Seconds"]],
+	relay1StatusAfterPowerFailure: [num:14, title:"Relay 1 Status After Power Failure", size:1, defaultVal:2, options:[0:"Forced off", 1:"Forced on", 2:"Restore previous state"]],
+	relay1LoadControl: [num:17, title:"Relay 1 Load Control", size:1, defaultVal:1, options:[0:"Disable Switch/ Enable Z-Wave", 1:"Enable Switch and Z-Wave", 2:"Disable Switch and Z-Wave"]],
+	relay1SwitchType: [num:20, title:"Relay 1 Switch Type", size:1, defaultVal:2, options:[0:"Toggle Switch", 1:"Momentary Light Switch", 2:"Toggle Up On/Down Off", 3:"3-way Impulse Control", 4:"Garage Door Mode"]],
+	relay1ImpulseDuration: [num:22, title:"Relay 1 Impulse Duration for 3-way", size:1, defaultVal:10, range:"2..200", desc:"2..200"],
+	relay2AutoOff: [num:5, title:"Relay 2 Auto Off Timer", size:2, defaultVal:0, range:"0..65535", desc:"0(disabled), 1..65535(timer unit)"],
+	relay2AutoOn: [num:6, title:"Relay 2 Auto On Timer", size:2, defaultVal:0, range:"0..65535", desc:"0(disabled), 1..65535(timer unit)"],
```

Comment:
Are those params really needed ? There are ST automations than might cover this functionality.
@greens ?

Reply:
Having the firmware handle it ensures that it continues to work when ST is experiencing issues..